backtrack/permutation.go

```diff
@@ -0,0 +1,26 @@
+// permutation.go
+// description: generate permutations of a given set
+// details:
+// In mathematics, a permutation of a set is, loosely speaking, an arrangement of its members into a sequence or linear order, or if the set is already ordered, a rearrangement of its elements. The word "permutation" also refers to the act or process of changing the linear order of an ordered set.
+// author: [sixwaaaay](https://github.com/sixwaaaay)
+// see permutation_test.go
+
+package backtrack
+
+func Permutation(nums []int) (set [][]int) {
+	n := len(nums)
+	var backtrack func(int)
+	backtrack = func(depth int) {
+		if depth == n {
+			set = append(set, append([]int{}, nums...))
+			return
+		}
+		for i := depth; i < len(nums); i++ {
+			nums[depth], nums[i] = nums[i], nums[depth]
+			backtrack(depth + 1)
+			nums[depth], nums[i] = nums[i], nums[depth]
+		}
+	}
+	backtrack(0)
+	return
```

Comment:
no need for this return

Reply:
when remove this return: .\permutation.go:26:1: missing return at end of function. so this return can't be removed.

Reply:
Oh, I didn't see the return `set` in the function signature. I think you should remove the `set` from the function return because it is very easy to miss that.

Having an explicit is not going to hurt the implementation and explicit is better than implicit.